clients/mobile/src/userWorkflow/account/hooks/useAccount.ts

```diff
@@ -0,0 +1,104 @@
+import { useNavigation } from '@react-navigation/native'
+import { useFetchClient } from '../../../setup/clients/useFetchClient'
+import { useEffect, useState } from 'react'
+import { useAuth } from '../../../authentication/context/useAuth'
+import { Platform } from 'react-native'
+import { Cache } from 'aws-amplify/utils'
+import { signOut } from 'aws-amplify/auth'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { SCREENS } from '../../../setup/constant/screens'
+import { FetchResponse } from '../../../setup/clients/FetchClient'
+import { AccountScreenNavigationProp } from '../../../setup/navigation/navigationTypes'
+
+interface User {
+  name: string;
+  location: string;
+}
+
+interface UserResponseData {
+  success: boolean;
+  data: {
+    phoneNumbers: string[];
+    name: string;
+    bloodGroup: string;
+    lastDonationDate: string;
+    height: number;
+    weight: number;
+    gender: string;
+    dateOfBirth: string;
+    availableForDonation: string;
+    NIDFront: string;
+    NIDBack: string;
+    lastVaccinatedDate: string;
+    preferredDonationLocations: Array<{
+      area: string;
+      city: string;
+      latitude: number;
+      longitude: number;
+    }>;
+  };
+  message: string;
+}
+
+interface UseAccountReturnType {
+  userData: User | null;
+  loading: boolean;
+  error: string | null;
+  handleSignOut: () => Promise<void>;
+}
+
+export const useAccount = (): UseAccountReturnType => {
+  const auth = useAuth()
+  const fetchClient = useFetchClient()
+  const navigation = useNavigation<AccountScreenNavigationProp>()
+  const [userData, setUserData] = useState<User | null>(null)
+  const [loading, setLoading] = useState<boolean>(true)
+  const [error, setError] = useState<string | null>(null)
+
+  const clearAllStorage = async(): Promise<void> => {
+    if (Platform.OS !== 'web') {
+      await AsyncStorage.clear()
+    }
+  }
+
+  const handleSignOut = async(): Promise<void> => {
+    try {
+      await auth.logoutUser()
+      await Promise.all([Cache.clear(), clearAllStorage()])
+      await signOut()
+      navigation.navigate(SCREENS.WELCOME)
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error(`Error: ${error.message}`)
+      } else {
+        throw new Error('Unknown error occurred during sign out')
+      }
+    }
+  }
+
+  const fetchUserData = async(): Promise<void> => {
+    setLoading(true)
+
+    try {
+      const response: FetchResponse<UserResponseData> = await fetchClient.get('/users')
+
+      if (response.status !== 200) {
+        throw new Error(`Error: ${response.status} ${response.message ?? 'Unknown error'}`)
+      }
+
+      const { name, preferredDonationLocations } = response.data
+      const location = `${preferredDonationLocations[0].city}, ${preferredDonationLocations[0].area}`
```

Comment:
What if `preferredDonationLocations` is empty?

Reply:
Updated the code, now checking if length of it before grabbing it's values